### DISCUSSION TITLE
Customize RTSP Stream

### DISCUSSION THREAD

**Discussion by Indianb0y016** [roleoroleo/yi-hack-Allwinner#436] [Ideas]
I realize that the limitations of the camera are going to be something that prevent people from really utilizing it in a high stability situations, but Im sure we can also agree that throwing out perfectly functioning cameras isnt friendly either. That being said, Im wondering if its possible to customize the rtsp stream. Parameters such as resolution (ex, high preset streaming at 720p rather an 1080p), Framerate (ex. low stream can be 5 and high can be 10-20 or so), and possibly codec choice. I doubt these cameras stream at h265, but I did see references to that in the source code. 

Are any of these possible with the hack already flashed? Or can they be implemented if I compile the project manually? Configuration doesnt have to be through webui, it can be through ssh. Thanks!

**Comment by roleoroleo** [comment]
The configuration of the streams is made by rmm process that uses API function available in the SDK,
But I'm not able to change this binary because the source code is not open.
Furthermore, the goal of this project is to maintain the features of the original firmware.
I didn't find a way to change resolution and/or fps.
But, with Allwinner-v2, I found a method to change the codec from h264 to hevc and viceversa writing a byte inside the flash memory.
https://github.com/roleoroleo/yi-hack-Allwinner-v2/issues/484
I don't remember if it is also available for Allwinner but I can check it.
